build-release.sh

```diff
@@ -4,46 +4,58 @@ set -eEuo pipefail
 
 eval "$(go env)"
 
-OUT_DIR="${1-dist}"
+set -x
+export OUT_DIR="${1-dist}"
 # To override the latest git tag as the version, pass something else as the second arg.
-VERSION=${2:-$(git describe --tags --always --dirty)}
+export VERSION=${2:-$(git describe --tags --always --dirty)}
 
 # To overwrite the version details, pass something as the third arg. Empty string disables it.
-VERSION_DETAILS=${3-"$(date -u +"%FT%T%z")/$(git describe --always --long --dirty)"}
+export VERSION_DETAILS=${3-"$(date -u +"%FT%T%z")/$(git describe --always --long --dirty)"}
+set +x
 
 build() {
     local ALIAS="$1" SUFFIX="${2}"  # Any other arguments are passed to the go build command as env vars
     local NAME="k6-${VERSION}-${ALIAS}"
 
-    local BUILD_ENV=("${@:3}")
     local BUILD_ARGS=(-o "${OUT_DIR}/${NAME}/k6${SUFFIX}" -trimpath)
-
     if [ -n "$VERSION_DETAILS" ]; then
         BUILD_ARGS+=(-ldflags "-X go.k6.io/k6/lib/consts.VersionDetails=$VERSION_DETAILS")
     fi
 
-    echo "- Building platform: ${ALIAS} (" "${BUILD_ENV[@]}" "go build" "${BUILD_ARGS[@]}" ")"
+    local PACKAGE_FORMATS
+    IFS="," read -ra PACKAGE_FORMATS <<< "${3}"
+
+    local ENV_VARS
+    IFS="," read -ra ENV_VARS <<< "${4}"
+
+    echo "- Building platform: ${ALIAS} (" "${ENV_VARS[@]}" "go build" "${BUILD_ARGS[@]}" ")"
 
     mkdir -p "${OUT_DIR}/${NAME}"
 
     # Subshell to not mess with the current env vars or CWD
     (
-        export "${BUILD_ENV[@]}"
+        export "${ENV_VARS[@]}"
         # Build a binary
-         go build "${BUILD_ARGS[@]}"
+        go build "${BUILD_ARGS[@]}"
+
+        for format in "${PACKAGE_FORMATS[@]}"; do
+            package "$format"
+        done
     )
 }
 
 package() {
-    local ALIAS="$1" FMT="$2"
-    local NAME="k6-${VERSION}-${ALIAS}"
+    local FMT="$1"
     echo "- Creating ${NAME}.${FMT} package..."
     case $FMT in
     deb|rpm)
-        # The go-bin-* tools expect the binary in /tmp/
-        [ ! -r /tmp/k6 ] && cp "${OUT_DIR}/${NAME}/k6" /tmp/k6
-        "go-bin-${FMT}" generate --file "packaging/${FMT}.json" -a amd64 \
-            --version "${VERSION#v}" -o "${OUT_DIR}/k6-${VERSION}-linux-amd64.${FMT}"
+        # nfpm can't substitute env vars in file paths, so we have to cd...
+        cd "${OUT_DIR}/${NAME}"
+        set -x
```

Comment:
Maybe a comment on why this is done specifically for nfpm ?

Reply:
set -x just shows the commands that are being executed, so it's easier to debug if there is a problem, not sure it deserves its own comment

Reply:
Why are we not doing this for the whole script then ? But specifically for this one part. I would argue it either should be everywhere or nowhere and if someone needs to debug something they can add it

Reply:
Resolved by the [latest force push](https://github.com/grafana/k6/compare/7bf212fa06fa9dbeb4f7ce7baee935445e1b878c..ae28e070b23f315cab179da884814f2a7ee22d18)